Lib/fontTools/feaLib/parser.py

```diff
@@ -826,8 +826,15 @@ def parse_substitute_(self):
                 'is not supported',
                 location)
 
+        # If there are remaining glyphs to parse, this is an invalid GSUB statement
+        if len(new) != 0:
+            raise FeatureLibError(
+                'This is an invalid GSUB statement. Make sure your statement '
+                'follows the format of one of the GSUB rule.',
```

Comment:
I still think the message is still quite verbose and the second sentence doesn't add much info to the fact that the statement is invalid. I'd simply write "Invalid substitution statement" (then the parser will add the location within the file and a user can debug the issue).